File: Array and Hashing/golang/two-sum.go
// https://leetcode.com/problems/two-sum/
func twoSum(nums []int, target int) []int {
    res := make([]int, 0, 2)
    n := len(nums)
    mp := make(map[int]int, n)
    for idx, val := range nums{
        v, ok := mp[val]
        if ok{
            res = append(res, v, idx)
            break 
        }else{
            mp[target-val] = idx
        }
    }
    return res
}

File: Array and Hashing/C++/two-sum.cpp
// https://leetcode.com/problems/two-sum/
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> x;
        int size = nums.size();
        for (int i = 0; i < size; i++)
        {
            auto t = x.find(nums[i]);
            if(t != x.end())
            {
                return {i, t->second};
            }
            else x[target - nums[i]] = i;
        }
        return {0, 0};
    }
};
